lib/operations/cursor_ops.js

```diff
@@ -20,39 +20,30 @@ function count(cursor, applySkipLimit, opts, callback) {
     if (typeof cursor.cursorSkip() === 'number') opts.skip = cursor.cursorSkip();
     if (typeof cursor.cursorLimit() === 'number') opts.limit = cursor.cursorLimit();
   }
+  if (cursor.s.cmd.hint) {
+    opts.hint = cursor.s.cmd.hint;
```

Comment:
I think we should seriously consider why this wasn't being applied in one case, and is in the other. Arguably this should also be moved to the count builder.

Reply:
I'm confused by this because it seems like the old code was setting `hint = cursor.s.cmd.hint` if there `cursor.s.cmd.hint` was set, but then overwriting `hint` with `cursor.s.options.hint` if that was set.